weblate/api/serializers.py

```diff
@@ -478,6 +487,25 @@ def to_internal_value(self, data):
             result["source_language"] = Language.objects.get(
                 code=result["source_language"]["code"]
             )
+        if "docfile" in result:
+            fake = create_component_from_doc(result)
+            format_cls = FILE_FORMATS[result["file_format"]]
+            LocalRepository.from_files(
+                fake.full_path,
+                {result["template"]: format_cls.get_new_file_content()},
+            )
```

Comment:
Is this really needed? The local repository is already created in `create_component_from_doc` and creating template with default content should not be needed.

```suggestion
```

Reply:
Without the template file content defined, the creation from doc fails. I checked without this, but then creation would just fail.

Reply:
Yes, the template has to be specified here. But either it should be specified by user or filled in automatically (see https://github.com/WeblateOrg/weblate/pull/4553/files#r496475217). Putting nonsense value will certainly not make it work (you put file content where filename is expected).

The major issue here is that validation apparently doesn't work as it should as the component creation happily accepts nonsense values.

Reply:
The problem is template value you pass in the tests. In UI the creation process is two step - first the uploaded file is imported to the repo, then normal component configuration follows where user selects the file to translate and that sets all the component parameters. To make the API simple, I'd make it just single step, but that means that the params has to match the uploaded file, that's what I've tried to do in the code suggestions (but I might have missed something as I didn't test it).